src/input.rs

```diff
@@ -626,58 +615,59 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
                         self.ctx.mouse_mut().scroll_px = 0;
                     },
                     TouchPhase::Moved => {
-                        let (_x, y): (i32, i32) = lpos.into();
-                        self.ctx.mouse_mut().scroll_px += y;
-                        let height = self.ctx.size_info().cell_height as i32;
-
-                        while self.ctx.mouse().scroll_px.abs() >= height {
-                            let code = if self.ctx.mouse().scroll_px > 0 {
-                                self.ctx.mouse_mut().scroll_px -= height;
-                                64
-                            } else {
-                                self.ctx.mouse_mut().scroll_px += height;
-                                65
-                            };
-
-                            self.scroll_terminal(code, modifiers, 1)
-                        }
+                        self.scroll_terminal(modifiers, lpos.y as i32);
                     },
                     _ => (),
                 }
             }
         }
     }
 
-    fn scroll_terminal(&mut self, code: u8, modifiers: ModifiersState, scroll_multiplier: u8) {
-        debug_assert!(code == 64 || code == 65);
-
-        let mouse_modes = TermMode::MOUSE_REPORT_CLICK | TermMode::MOUSE_DRAG | TermMode::MOUSE_MOTION;
+    fn scroll_terminal(&mut self, modifiers: ModifiersState, new_scroll_px: i32) {
+        let mouse_modes =
+            TermMode::MOUSE_REPORT_CLICK | TermMode::MOUSE_DRAG | TermMode::MOUSE_MOTION;
+        let height = self.ctx.size_info().cell_height as i32;
 
         // Make sure the new and deprecated setting are both allowed
-        let faux_scrolling_lines = self.mouse_config
+        let faux_multiplier = self
+            .mouse_config
             .faux_scrollback_lines
             .unwrap_or(self.scrolling_config.faux_multiplier as usize);
 
+        self.ctx.mouse_mut().scroll_px += new_scroll_px;
```

Comment:
Hi,
This makes the value of `scroll_px` become `(scroll_px + new_scroll_px) * multiplier`, which is not correct. 
The correct value should be `scroll_px + (new_scroll_px * multiplier)`.